app/Http/Requests/Album/SetAlbumCoverRequest.php

```diff
@@ -22,8 +23,7 @@ class SetAlbumCoverRequest extends BaseApiRequest implements HasAlbum, HasPhoto
 	 */
 	public function authorize(): bool
 	{
-		return $this->authorizeAlbumWrite($this->album) &&
-			$this->authorizePhotoVisible($this->photo);
+		return $this->authorizeAlbumWrite($this->album) && Gate::check('access', $this->photo);
```

Comment:
So I saw in a number of places that `access` replaced both `Accessible` and `Visible`. This seems suspicious on the first sight, although I don't have anymore a good enough mental image of all these permissions to figure out if that's correct or not. @nagmat84 will have to weigh in...

Reply:
Consider this from the point of view of the User: "Can `current user` `access` this `album`/`photo`?"

Reply:
For photos it should make no difference: every visible photo is accessible. For password-protected albums though, there's definitely a difference between an album being visible and being accessible. Now that I think about it though, since in this line you test for the visibility of a photo, you should be fine. I haven't checked if that was the case in all other instances as well?

Reply:
Both `AlbumAuthorizationProvider` and `PhotoAuthorizationProvider` contain long explanations in form of comments which define all the terms in a very precise way. Also see https://github.com/LycheeOrg/Lychee/pull/1403#discussion_r930160704

The term "accessible" has not been defined for photos. Photos can be

 - visible
 - downloadable
 - editable
 - searchable

Albums can be

 - visible
 - accessible
 - reachable
 - browsable
 - editable

I just do not feel like repeating all the definitions here even though it would only mean copy & paste from the source code. For each property the source code contains comments a long the lines

> A photo is called visible if and only if <list of conditions>.

> Sometimes the condition is short like in

> An album is called reachable if and only if it is visible _and_ accessible.

and both terms _visible_ and _accessible_ have been defined before.

Back to the original question. "Accessible" is not defined for photos. Unless @ildyria  does not intend to rewrite all these definitions, comments and also all other occurrences in the code elsewhere, I would strongly argue to keep the current wording and not to start using new terms in a different meaning than now.

One of the former problems was that Lychee used the same term for different things, but also different terms for the same thing. It was a lot of work to fix this problem in both directions: on the one hand to ensure that whenever a term is used it always means exactly the same no matter of the context; on the other hand to ensure that whenever the same thing is meant it is referred to by the same term always.

Introducing a new term for the same thing will soon lead to the previous unfortunate situation.

Reply:
Will fix that.

Reply:
should be fixed.

Reply:
Not entirely. You still have `PhotoPolicy::ACCESS` and "access" is an undefined term for photos.

Reply:
Done. https://github.com/LycheeOrg/Lychee/pull/1403/commits/36ee767c016af2f031093bf0a29ee3a956a2e282